core/server/master/src/test/java/alluxio/master/meta/AlluxioMasterRestServiceHandlerTest.java

```diff
@@ -299,4 +305,57 @@ public void isMounted() {
     assertFalse(handler.isMounted(hdfsUri));
     assertFalse(handler.isMounted(MetricsSystem.escape(new AlluxioURI(hdfsUri))));
   }
+
+  @Test
+  public void testGetWebUILogsByRegex() throws IOException {
+    File logsDir = mTestFolder.newFolder("logs");
+    logsDir.mkdirs();
+    String[] fileArray0 = new String[] {
+        "master.log",
+        "master.log.1",
+        "master.log.100",
+        "master.log",
+        "master.log.1",
+        "master.log.100",
+        "master.txt",
+        "alluxio-master-exit-metrics-20230526-085548.json"
+    };
+    String[] fileArray1 = new String[] {
+        "master.log.a",
+        "master.loga",
+        "master.bin",
+    };
+
+    for (String fileName : fileArray0) {
+      File file0 = new File(logsDir, fileName);
+      file0.createNewFile();
+    }
+    for (String fileName : fileArray1) {
+      File file0 = new File(logsDir, fileName);
+      file0.createNewFile();
+    }
+
+    Configuration.set(PropertyKey.LOGS_DIR, logsDir.getPath());
+    FileSystemMaster mockMaster = mock(FileSystemMaster.class);
+    BlockMaster mockBlockMaster = mock(BlockMaster.class);
+
+    AlluxioMasterProcess masterProcess = PowerMockito.mock(AlluxioMasterProcess.class);
+    when(masterProcess.getMaster(FileSystemMaster.class)).thenReturn(mockMaster);
+    when(masterProcess.getMaster(BlockMaster.class)).thenReturn(mockBlockMaster);
+    when(mockBlockMaster.getGlobalStorageTierAssoc()).thenReturn(
+        new DefaultStorageTierAssoc(
+            PropertyKey.MASTER_TIERED_STORE_GLOBAL_LEVELS,
+            PropertyKey.Template.MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS));
+
+    ServletContext context = mock(ServletContext.class);
+    when(context.getAttribute(MasterWebServer.ALLUXIO_MASTER_SERVLET_RESOURCE_KEY)).thenReturn(
+        masterProcess);
+    AlluxioMasterRestServiceHandler handler = new AlluxioMasterRestServiceHandler(context);
+    Response response = handler.getWebUILogs("", "0", "", "20");
+    Assert.assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+    List<UIFileInfo> fileInfos = ((MasterWebUILogs) response.getEntity()).getFileInfos();
+    String[] actualFileNameArray =
+        fileInfos.stream().map(fileInfo -> fileInfo.getName()).toArray(String[]::new);
+    Arrays.equals(fileArray0, actualFileNameArray);
```

Comment:
is the list order guaranteed? Use Set equality instead of Array equality?

Reply:
Done

Reply:
Done